routes/api.php

```diff
@@ -20,6 +22,13 @@
 Route::post('/signup', RegisterUser::class)->name('signup');
 Route::post('/login', LoginController::class)->name('login');
 
-Route::middleware('auth:sanctum')->get('/user', function (Request $request) {
-    return $request->user();
+Route::middleware(['auth:sanctum'])->group(function () {
+    Route::get('/user', function (Request $request) { return $request->user(); });
+    Route::post('/logout', LogoutController::class)->name('logout');
 });
+
+Route::any('{any}', function(){
```

Comment:
Esto lo agregue porque cuando intentaba acceder a alguna de las rutas que necesitabas autenticacion me tiraba un error laravel. No me queda claro si es la mejor forma de hacerlo o no

Reply:
Que error exactamente te tira?

Reply:
@gianfranco-rocco 
Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException: The GET method is not supported for route api/login. Supported methods: POST. in file /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php on line 122

Reply:
Claro lo que sucede es que estas intentando acceder a una ruta la cual no tiene ninguna definicion GET, por lo tanto sucede eso. Al ser esto una API no se si tiene sentido tener un default para rutas no encontradas, eso es algo que tiene mas sentido en el front dado que si no encontras una ruta o bien redirigis al usuario al home, o le mostrar una pantalla indicandole que no existe tal ruta. Pero para las APIs yo en lo personal no lo haria. No se que opina Sergio.

Reply:
Esta correcto, igualmente se puede configurar toda la app para que te devuelva siempre las respuestas en JSON y no en HTML

Reply:
Lo resolvi aceptando en postman solamente json en el header de la consulta